packages/shared/lib/core/layer-2/utils/getLayer2MetadataForTransfer.ts

```diff
@@ -33,68 +20,9 @@ export function getLayer2MetadataForTransfer(transactionDetails: NewTransactionD
     const parameters = encodeSmartContractParameters(smartContractParameters)
     metadataStream.writeBytes('smartContractParameters', parameters.length, parameters)
 
-    const allowance = encodeAllowance(transactionDetails)
+    const allowance = encodeAssetAllowance(transactionDetails)
     metadataStream.writeBytes('allowance', allowance.length, allowance)
 
```

Comment:
Do we no longer need the end byte?

Reply:
At the time I was implementing this, the specification didn't include the NFTs. It turns out that the ending signal byte represents that no NFTs have been added to the transaction. This change is now reflected on the line below and I just use the empty buffer constant: https://github.com/iotaledger/firefly/blob/d74e25a7d1edb045dbaa19f744b8f9172fc6eff5/packages/shared/lib/core/layer-2/helpers/encodeAssetAllowance.ts#L27